src/TestFramework/MSTest.Core/Assertions/Assert.IsInstanceOfType.cs

```diff
@@ -116,6 +136,17 @@ public static void IsInstanceOfType(object value, Type expectedType, [CallerArgu
         }
     }
 
+    /// <summary>
+    /// Tests whether the specified object is an instance of the generic
+    /// type and throws an exception if the generictype is not in the
+    /// inheritance hierarchy of the object.
+    /// </summary>
+    public static void IsInstanceOfType<T>(object value, [CallerArgumentExpression("value")] string message = null, params object[] parameters)
```

Comment:
I will make some test later but I suspect there is a problem with such API (not related to your change). You will be forced to pass a message when you want to use parameters which means that we wouldn't be using the `CallerArgumentExpression`

Reply:
I confirm that this is not a good pattern. We cannot combine `CallerArgumentExpressionAttribute` and `params`.